File: examples/python/example_src/simple.py
"""
Simple example Python functions for QA Agent test generation.

These functions demonstrate basic arithmetic operations that are simple to test.
"""

def add_numbers(a, b):
    """
    Add two numbers together.
    
    Args:
        a: First number
        b: Second number
        
    Returns:
        Sum of a and b
    """
    return a + b

def subtract_numbers(a, b):
    """
    Subtract b from a.
    
    Args:
        a: First number
        b: Second number
        
    Returns:
        Difference of a and b
    """
    return a - b

def multiply_numbers(a, b):
    """
    Multiply two numbers.
    
    Args:
        a: First number
        b: Second number
        
    Returns:
        Product of a and b
    """
    return a * b

def divide_numbers(a, b):
    """
    Divide a by b.
    
    Args:
        a: First number (dividend)
        b: Second number (divisor)
        
    Returns:
        Quotient of a divided by b
        
    Raises:
        ZeroDivisionError: If b is zero
    """
    if b == 0:
        raise ZeroDivisionError("Cannot divide by zero")
    return a / b

File: examples/js/example_src/simple.js
/**
 * Simple example JavaScript functions for QA Agent test generation.
 * 
 * These functions demonstrate basic arithmetic operations that are simple to test.
 */

/**
 * Add two numbers together.
 * 
 * @param {number} a - First number
 * @param {number} b - Second number
 * @returns {number} Sum of a and b
 */
function addNumbers(a, b) {
  return a + b;
}

/**
 * Subtract b from a.
 * 
 * @param {number} a - First number
 * @param {number} b - Second number
 * @returns {number} Difference of a and b
 */
function subtractNumbers(a, b) {
  return a - b;
}

/**
 * Multiply two numbers.
 * 
 * @param {number} a - First number
 * @param {number} b - Second number
 * @returns {number} Product of a and b
 */
function multiplyNumbers(a, b) {
  return a * b;
}

/**
 * Divide a by b.
 * 
 * @param {number} a - First number (dividend)
 * @param {number} b - Second number (divisor)
 * @returns {number} Quotient of a divided by b
 * @throws {Error} If b is zero
 */
function divideNumbers(a, b) {
  if (b === 0) {
    throw new Error("Cannot divide by zero");
  }
  return a / b;
}

// Export functions for use in other modules
module.exports = {
  addNumbers,
  subtractNumbers,
  multiplyNumbers,
  divideNumbers
};